src/test/java/org/jooq/lambda/SeqTest.java

```diff
@@ -1543,6 +1543,22 @@ public void testIterate() {
     }
 
     @Test
+    public void testIterateWhilePresent() {
+        assertEquals(
+              asList(),
+              Seq.iterateWhilePresent(null, i -> Optional.empty()).toList()
+        );
```

Comment:
Well, as to why call `iterateWhilePresent` when you *know* that the resulting `Seq` will be empty - it's for the same reason why you call `Seq.seq(Optional)` instead of `optional.isPresent() ? Seq.of(optional.get()) : Seq.empty()`, i.e. for convenience.

But I guess `seed` usually is non-empty so I agree with you that in this particular case consistency with the existing `iteratate()` is more important than robustness.

Solution 1 (`Objects.requireNonNull(seed)`) it is, then, right?